### ISSUE TITLE
Integrate travis to run tests and linting

### ISSUE THREAD

**Issue by syxolk** [Rostlab/JS18_ProjectA_Group2#6]
Add `.travis.yml` to the project and enable Travis CI on the project. Run `npm run test` in travis.

**Comment by syxolk** [comment]
`npm test` is failing because `tslint` exits with status 2.

I won't configure travis for you if the unit tests are failing.

This is the entire error message from tslint:
```
src/app/app.module.ts[38, 3]: comment must start with a space
src/app/app.routes.ts[4, 3]: comment must start with a space
src/app/home/home.component.ts[10, 2]: file should end with a newline
src/app/home/home.component.ts[9, 1]: trailing whitespace
src/app/posts/edit-post/edit-post.component.ts[26, 5]: parameters are not aligned
src/app/posts/edit-post/edit-post.component.ts[27, 5]: parameters are not aligned
src/app/posts/edit-post/edit-post.component.ts[28, 5]: parameters are not aligned
src/app/posts/edit-post/edit-post.component.ts[49, 8]: statements are not aligned
src/app/posts/edit-post/edit-post.component.ts[41, 44]: Parentheses are required around the parameters of an arrow function definition
src/app/posts/edit-post/edit-post.component.ts[54, 5]: if statements must be braced
src/app/posts/edit-post/edit-post.component.ts[19, 2]: The class property 'form' must be marked either 'private', 'public', or 'protected'
src/app/posts/edit-post/edit-post.component.ts[21, 3]: Declaration of public instance field not allowed after declaration of private instance field. Instead, this should come at the beginning of the class/interface.
src/app/posts/edit-post/edit-post.component.ts[22, 3]: Declaration of public instance field not allowed after declaration of private instance field. Instead, this should come at the beginning of the class/interface.
src/app/posts/edit-post/edit-post.component.ts[12, 1]: Consecutive blank lines are forbidden
src/app/posts/edit-post/edit-post.component.ts[54, 26]: trailing whitespace
src/app/posts/edit-post/edit-post.component.ts[70, 27]: trailing whitespace
src/app/posts/edit-post/edit-post.component.ts[46, 20]: Unnecessarily quoted property 'id' found.
src/app/posts/edit-post/edit-post.component.ts[59, 9]: Unnecessarily quoted property 'id' found.
src/app/posts/edit-post/edit-post.component.ts[60, 9]: Unnecessarily quoted property 'data' found.
src/app/posts/edit-post/edit-post.component.ts[61, 11]: Unnecessarily quoted property 'title' found.
src/app/posts/edit-post/edit-post.component.ts[62, 11]: Unnecessarily quoted property 'content' found.
src/app/posts/edit-post/edit-post.component.ts[46, 20]: " should be '
src/app/posts/edit-post/edit-post.component.ts[59, 9]: " should be '
src/app/posts/edit-post/edit-post.component.ts[60, 9]: " should be '
src/app/posts/edit-post/edit-post.component.ts[61, 11]: " should be '
src/app/posts/edit-post/edit-post.component.ts[62, 11]: " should be '
src/app/posts/edit-post/edit-post.component.ts[40, 22]: Missing semicolon
src/app/posts/edit-post/edit-post.component.ts[39, 10]: Implement lifecycle hook interface OnInit for method ngOnInit in class EditPostComponent (https://goo.gl/w1Nwk3)
src/app/posts/graphql/mutations.ts[26, 3]: file should end with a newline
src/app/posts/graphql/queries.ts[21, 3]: file should end with a newline
src/app/posts/graphql/queries.ts[3, 32]: missing whitespace
src/app/posts/graphql/schema.ts[29, 2]: file should end with a newline
src/app/posts/graphql/schema.ts[6, 4]: Missing semicolon
src/app/posts/graphql/schema.ts[13, 4]: Missing semicolon
src/app/posts/graphql/schema.ts[28, 4]: Missing semicolon
src/app/posts/graphql/schema.ts[3, 8]: expected onespace after colon in property-declaration
src/app/posts/graphql/schema.ts[4, 11]: expected onespace after colon in property-declaration
src/app/posts/graphql/schema.ts[5, 13]: expected onespace after colon in property-declaration
src/app/posts/graphql/schema.ts[11, 8]: expected onespace after colon in property-declaration
src/app/posts/graphql/schema.ts[12, 11]: expected onespace after colon in property-declaration
src/app/posts/graphql/schema.ts[24, 10]: expected onespace after colon in property-declaration
src/app/posts/graphql/schema.ts[3, 8]: missing whitespace
src/app/posts/graphql/schema.ts[4, 11]: missing whitespace
src/app/posts/graphql/schema.ts[5, 13]: missing whitespace
src/app/posts/graphql/schema.ts[11, 8]: missing whitespace
src/app/posts/graphql/schema.ts[12, 11]: missing whitespace
src/app/posts/graphql/schema.ts[24, 10]: missing whitespace
src/app/posts/index.ts[1, 46]: file should end with a newline
src/app/posts/new-post/new-post.component.ts[32, 5]: if statements must be braced
src/app/posts/new-post/new-post.component.ts[16, 3]: The class property 'form' must be marked either 'private', 'public', or 'protected'
src/app/posts/new-post/new-post.component.ts[9, 1]: Consecutive blank lines are forbidden
src/app/posts/new-post/new-post.component.ts[49, 26]: trailing whitespace
src/app/posts/new-post/new-post.component.ts[36, 9]: Unnecessarily quoted property 'data' found.
src/app/posts/new-post/new-post.component.ts[37, 11]: Unnecessarily quoted property 'title' found.
src/app/posts/new-post/new-post.component.ts[38, 11]: Unnecessarily quoted property 'content' found.
src/app/posts/new-post/new-post.component.ts[36, 9]: " should be '
src/app/posts/new-post/new-post.component.ts[37, 11]: " should be '
src/app/posts/new-post/new-post.component.ts[38, 11]: " should be '
src/app/posts/new-post/new-post.component.ts[38, 22]: missing whitespace
src/app/posts/post-detail/post-detail.component.ts[42, 48]: Parentheses are required around the parameters of an arrow function definition
src/app/posts/post-detail/post-detail.component.ts[26, 14]: Class name must be in pascal case
src/app/posts/post-detail/post-detail.component.ts[55, 2]: file should end with a newline
src/app/posts/post-detail/post-detail.component.ts[52, 5]: The class method 'ngOnDestroy' must be marked either 'private', 'public', or 'protected'
src/app/posts/post-detail/post-detail.component.ts[31, 5]: Declaration of public instance field not allowed after declaration of private instance field. Instead, this should come at the beginning of the class/interface.
src/app/posts/post-detail/post-detail.component.ts[33, 5]: Declaration of public instance field not allowed after declaration of private instance field. Instead, this should come at the beginning of the class/interface.
src/app/posts/post-detail/post-detail.component.ts[35, 5]: Declaration of public instance field not allowed after declaration of private instance field. Instead, this should come at the beginning of the class/interface.
src/app/posts/post-detail/post-detail.component.ts[11, 1]: Consecutive blank lines are forbidden
src/app/posts/post-detail/post-detail.component.ts[19, 1]: Consecutive blank lines are forbidden
src/app/posts/post-detail/post-detail.component.ts[47, 26]: Unnecessarily quoted property 'id' found.
src/app/posts/post-detail/post-detail.component.ts[47, 26]: " should be '
src/app/posts/post-list/post-list.component.ts[35, 63]: Parentheses are required around the parameters of an arrow function definition
src/app/posts/post-list/post-list.component.ts[10, 1]: Consecutive blank lines are forbidden
src/app/posts/post-list/post-list.component.ts[48, 9]: Missing semicolon
src/app/posts/post.interface.ts[4, 2]: file should end with a newline
src/app/posts/posts-filter/posts-filter.pipe.ts[8, 61]: comment must start with a space
src/app/posts/posts-filter/posts-filter.pipe.ts[12, 2]: file should end with a newline
src/app/posts/posts-filter/posts-filter.pipe.ts[9, 1]: Exceeds maximum line length of 100
src/app/posts/posts-filter/posts-filter.pipe.ts[7, 1]: The class method 'transform' must be marked either 'private', 'public', or 'protected'
src/app/posts/posts-filter/posts-filter.pipe.ts[6, 54]: missing whitespace
src/app/posts/posts-filter/posts-filter.pipe.ts[7, 51]: missing whitespace
src/app/posts/posts-filter/posts-filter.pipe.ts[2, 40]: Missing semicolon
src/app/posts/posts-filter/posts-filter.pipe.ts[7, 44]: expected onespace after colon in call-signature
src/app/posts/posts-filter/posts-filter.pipe.ts[7, 16]: expected nospace before colon in parameter
src/app/posts/posts-filter/posts-filter.pipe.ts[7, 18]: expected onespace after colon in parameter
src/app/posts/posts-filter/posts-filter.pipe.ts[7, 36]: expected onespace after colon in parameter
src/app/posts/posts-filter/posts-filter.pipe.ts[9, 42]: expected onespace after colon in parameter
src/app/posts/posts-filter/posts-filter.pipe.ts[8, 13]: missing whitespace
src/app/posts/posts-filter/posts-filter.pipe.ts[8, 23]: missing whitespace
src/app/posts/posts-filter/posts-filter.pipe.ts[8, 53]: missing whitespace
src/app/posts/posts-filter/posts-filter.pipe.ts[9, 48]: missing whitespace
src/app/posts/posts-filter/posts-filter.pipe.ts[9, 50]: missing whitespace
src/app/posts/posts-filter/posts-filter.pipe.ts[9, 102]: missing whitespace
src/app/posts/posts-filter/posts-filter.pipe.ts[1, 23]: missing whitespace
src/app/posts/posts-filter/posts-filter.pipe.ts[4, 10]: missing whitespace
src/app/posts/posts-filter/posts-filter.pipe.ts[7, 18]: missing whitespace
src/app/posts/posts-filter/posts-filter.pipe.ts[7, 36]: missing whitespace
src/app/posts/posts-filter/posts-filter.pipe.ts[7, 44]: missing whitespace
src/app/posts/posts-filter/posts-filter.pipe.ts[8, 54]: missing whitespace
src/app/posts/posts-filter/posts-filter.pipe.ts[8, 59]: missing whitespace
src/app/posts/posts-filter/posts-filter.pipe.ts[9, 42]: missing whitespace
src/app/posts/posts-filter/posts-filter.pipe.ts[1, 8]: You need to leave whitespaces inside of the import statement's curly braces
src/app/posts/posts-filter/posts-filter.pipe.ts[2, 8]: You need to leave whitespaces inside of the import statement's curly braces
src/app/posts/posts-filter/posts-filter.pipe.ts[4, 5]: The name of the Pipe decorator of class PostsFilterPipe should be named camelCase with prefix my, however its value is "postsFilter".
src/app/posts/posts.module.ts[6, 9]: trailing whitespace
src/app/posts/posts.module.ts[19, 65]: Missing semicolon
src/app/posts/posts.module.ts[19, 8]: You need to leave whitespaces inside of the import statement's curly braces
src/app/posts/posts.routes.ts[18, 3]: file should end with a newline
src/app/posts/posts.routes.ts[6, 1]: Consecutive blank lines are forbidden
src/app/posts/posts.service.ts[25, 18]: Parentheses are required around the parameters of an arrow function definition
src/app/posts/posts.service.ts[59, 2]: file should end with a newline
src/app/posts/posts.service.ts[19, 5]: The class method 'get' must be marked either 'private', 'public', or 'protected'
src/app/posts/posts.service.ts[28, 5]: The class method 'delete' must be marked either 'private', 'public', or 'protected'
src/app/posts/posts.service.ts[58, 1]: Consecutive blank lines are forbidden
src/app/posts/posts.service.ts[34, 21]: Unnecessarily quoted property 'id' found.
src/app/posts/posts.service.ts[34, 21]: " should be '
src/app/posts/posts.service.ts[52, 27]: Missing semicolon
src/app/profile/index.ts[1, 37]: file should end with a newline
src/app/profile/profile.component.ts[24, 1]: Exceeds maximum line length of 100
src/app/profile/profile.component.ts[32, 1]: Exceeds maximum line length of 100
src/app/profile/profile.component.ts[38, 3]: The class property 'currentProfile' must be marked either 'private', 'public', or 'protected'
src/app/profile/profile.component.ts[44, 3]: The class method 'ngOnInit' must be marked either 'private', 'public', or 'protected'
src/app/profile/profile.component.ts[14, 24]: missing whitespace
src/app/profile/profile.component.ts[15, 10]: Missing semicolon
src/app/profile/profile.component.ts[41, 29]: Missing semicolon
src/graphql.client.ts[11, 2]: file should end with a newline
```

**Comment by syxolk** [comment]
npm test still fails:
```
> igraph@1.0.0 test /home/hans/workspace/JS18_ProjectA_Group2
> npm run tst -- --report lcovonly && cat ./coverage/lcov.info | coveralls


> igraph@1.0.0 tst /home/hans/workspace/JS18_ProjectA_Group2
> istanbul cover _mocha -- -R spec --recursive test "--report" "lcovonly"

Unable to resolve file [_mocha]
Try "istanbul help" for usage
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! igraph@1.0.0 tst: `istanbul cover _mocha -- -R spec --recursive test "--report" "lcovonly"`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the igraph@1.0.0 tst script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
```

**Comment by syxolk** [comment]
Make sure it works till tomorrow.

**Comment by jyotirmay123** [comment]
@iraklitchedia  has already resolved it. Please push the changes ASAP. Thank you.

**Comment by syxolk** [comment]
Fixed with #29
